PivotSubscriptions/Utility.cls

```diff
@@ -0,0 +1,191 @@
+Class PivotSubscriptions.Utility
+{
+
+/// Utility method for saving and updating global variables
+ClassMethod ConfigureSettings(pAdminEmail, pHttp, pServer, pPort) {
+	Set ^PivotSubscriptions.Settings("AdminEmail") = pAdminEmail
+	Set ^PivotSubscriptions.Settings("Http") = pHttp
+	Set ^PivotSubscriptions.Settings("Server") = pServer
+	Set ^PivotSubscriptions.Settings("Port") = pPort
+	
+	Quit $$$OK
+}
+
+ClassMethod CheckConfigurationGlobal(pConfigurationOptionName As %String)
+{
+	If (pConfigurationOptionName '= "") {
+		If (^PivotSubscriptions.Settings(pConfigurationOptionName) '= "") {
+			Quit 1
+		}
+	}
+
+	Quit 0
+}
+
+ClassMethod GenerateLink(ByRef link As %String) As %Status 

```

Comment:
Now that we have the settings to configure the URL, I think we can skip extracting parts of the URL returned by GetWebServerPort and start with a blank string. 
1) If ^PivotSubscriptions.Settings("Http") is populated, use that value. If it is not, just use "http://"
2) If ^PivotSubscriptions.Settings("Server") is populated, use that value. If not, use h
3) If ^PivotSubscriptions.Settings("Port") is populated, use that value. If not, use p

Reply:
h does not define the server, it just defines the current user or local machine. I will implement it instead to extract the server from the url

Reply:
Actually, it just works. I didn't expect it to, so I will just use h.